shell/utils/__tests__/formatter.test.ts

```diff
@@ -0,0 +1,143 @@
+          error: false, message: 'Resource is always ready', name: 'active', transitioning: false
+        },
+        uid: 'fcc4aa44-ce0c-4a7d-bd67-e77040b889a7'
+      }
+    },
+    {
+      id:    'test4',
+      type:  'secret',
+      links: {
+        remove: 'https://test4:8000', self: 'https://test4:8000', update: 'https://test4:8000', view: 'https://test4:8000'
+      },
+      _type:      'Opaque',
+      apiVersion: 'v1',
+      data:       { hash: 'test4' },
+      immutable:  true,
+      kind:       'Secret',
+      metadata:   {
+        annotations:       { 'field.cattle.io/projectId': 'test4' },
+        creationTimestamp: '2023-06-26T22:51:42Z',
+        fields:            ['Opaque', 1, '56d'],
+        name:              'test4',
+        namespace:         'test',
+        relationships:     null,
+        resourceVersion:   '4306',
+        state:             {
+          error: false, message: 'Resource is always ready', name: 'active', transitioning: false
+        },
+        uid: 'fcc4aa44-ce0c-4a7d-bd67-e77040b889a6'
+      }
+    },
+    {
+      id:    'test1',
+      type:  'secret',
+      links: {
+        remove: 'https://test1:8000', self: 'https://test1:8000', update: 'https://test1:8000', view: 'https://test1:8000'
+      },
+      _type:      'Custom',
+      apiVersion: 'v1',
+      data:       { hash: 'test1', 'encryption-provider-config.yaml': 'MTIzNFFhYWEh' },
+      immutable:  true,
+      kind:       'Secret',
+      metadata:   {
+        annotations:       { 'field.cattle.io/projectId': 'test' },
+        creationTimestamp: '2023-06-26T22:51:42Z',
+        fields:            [1, '56d'],
+        name:              'test1',
+        namespace:         'test',
+        relationships:     null,
+        resourceVersion:   '4306',
+        state:             {
+          error: false, message: 'Resource is always ready', name: 'active', transitioning: false
+        },
+        uid: 'fcc4aa44-ce0c-4a7d-bd67-e77040b889a9'
+      }
+    },
+    {
+      id:    'test3',
+      type:  'secret',
+      links: {
+        remove: 'https://test3:8000', self: 'https://test3:8000', update: 'https://test3:8000', view: 'https://test3:8000'
+      },
+      _type:      'Opaque',
+      apiVersion: 'v1',
+      data:       { hash: 'test', 'encryption-provider-config.yaml': 'MTIzNFFhYWEh' },
+      immutable:  true,
+      kind:       'Secret',
+      metadata:   {
+        annotations:       { 'field.cattle.io/projectId': 'test' },
+        creationTimestamp: '2023-06-26T22:51:42Z',
+        fields:            ['Opaque', 1, '56d'],
+        namespace:         'test',
+        relationships:     null,
+        resourceVersion:   '4306',
+        state:             {
+          error: false, message: 'Resource is always ready', name: 'active', transitioning: false
+        },
+        uid: 'fcc4aa44-ce0c-4a7d-bd67-e77040b889a6'
+      }
+    }];
+  const chart = 'test';
+  const testCases = [[chart, 2], ['test1', 0]];
```

Comment:
Unless there's a `describe` scope case, it is convenient to avoid unnecessary constants declaration. A direct definition of the value within the `each` seems more appropriate.